### ISSUE TITLE
TypeScript: references/rename does not work for properties in JSDoc comments

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/vscode#30195]
Using TypeScript is `checkJs` mode, when an identifier is selected inside a JSDoc comment

- the references are not highlighted
- "find references" and "rename" fail to work

You can see the absence of highlighted references in this example:

![image](https://user-images.githubusercontent.com/921609/27917793-fc13a810-6264-11e7-9f65-c2648caba89b.png)

You can verify the JSDoc comment is "doing something" by checking the type flows through:

![image](https://user-images.githubusercontent.com/921609/27917817-09ca3ae6-6265-11e7-85f0-4696c273c03a.png)

**Comment by mjbvz** [comment]
What version of TypeScript are you using? TypeScript 2.4 improved support for jsdocs. Using VSCode 1.14 (the current insiders build) I see this working properly in some simple cases:

![jul-06-2017 17-21-58](https://user-images.githubusercontent.com/12821956/27937948-a5420530-626f-11e7-93e9-cad521e9d08f.gif)

**Comment by OliverJAsh** [comment]
Ah yes, it works with 2.4. Thanks!

**Comment by andy-ms** [comment]
Hi, might need some more context to tell what exactly the problem is.
Currently defining generic classes is not supported in JS: Microsoft/TypeScript#16411

But I made an example by defining the generic superclass in a `.d.ts` file:

**C.d.ts**
```ts
export default class C<T> {
    t: T;
}
```

**D.js**
```js
import C from "./C";

/** @augments {C<{ x: number }>} */
class D extends C {
    constructor() {
        super();
        const { x } = this.t;
    }
}
```

And I successfully found all references to `x`. So there must be more to your situation that is preventing us from correctly finding references.
